packages/compass-saved-aggregations-queries/src/stores/open-item.ts

```diff
@@ -264,28 +404,35 @@ export const openSavedItem =
     }
 
     const { database, collection } = item;
+    const multiConnectionsEnabled =
+      preferencesAccess.getPreferences().enableNewMultipleConnectionSystem;
+
+    if (!multiConnectionsEnabled) {
+      // In single connections mode, we only expect one connections to be here
+      // in the list of active connections.
+      const [activeConnection] = activeConnections;
+      const { error, dataService, instance } = dispatch(
+        getDataServiceAndInstanceForConnection(activeConnection.id)
+      );
+      if (error) {
+        return;
+      }
```

Comment:
That's kind of an unexpected pattern, if you really want to use the action to get those values, you can throw right in the get method, no need to return an error to re-throw it, but also see my other comment: every place where we get those, we throw if value is not available, I feel like we should just make the getter method account for that so that we don't need to check everywhere